app/main.py

```diff
@@ -1,34 +1,93 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.cell = (row, column)
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+        self.decks = []
+        for row_index in range(start[0], end[0] + 1):
+            for column_index in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row_index, column_index))
+        self.is_drowned = is_drowned
+
+    def get_deck(self, row: int, column: int) -> tuple:
+        for _deck in self.decks:
+            if (row, column) in _deck.cell:
+                return _deck.cell
+
+    def fire(self, row: int, column: int) -> None:
+        if (row, column) in self.decks:
+            if not self.decks:
+                self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+
+    def __init__(self, ships: list) -> None:
+        self.field = {}
+        for _ship in ships:
+            self.field[_ship] = Ship(_ship[0], _ship[1])
```

Comment:
```suggestion
            self.field[ship] = Ship(ship[0], ship[1])
```
Why you need these in all places???